inference/explained_bool.go

```diff
@@ -29,6 +30,8 @@ import (
 type ExplainedBool interface {
 	Val() bool
 	String() string
+	getPrimitiveFullTrigger() primitiveFullTrigger
+	getExplainedBool() ExplainedBool
```

Comment:
```suggestion
	DeeperReason() ExplainedBool
```
Right? because `ExplainedBool.getExplainedBool` reads a little weird to me ;)

Reply:
Good suggestion. I just un-exported `deeperReason()` since it is only used locally from within the pakcage.